server/src/main/java/org/candlepin/model/AsyncJobStatusCurator.java

```diff
@@ -28,6 +37,54 @@ public AsyncJobStatusCurator() {
         super(AsyncJobStatus.class);
     }
 
-    // TODO: Add things here as needed
+    /**
+     * Fetches a collection of jobs in the given states. If no jobs can be found in the states
+     * specified, this method returns an empty collection.
+     *
+     * @param states
+     *  a collection of states to use for filtering jobs
+     *
+     * @return
+     *  a collection of jobs in the provided states
+     */
+    public List<AsyncJobStatus> getJobsInState(Collection<AsyncJobStatus.JobState> states) {
+        if (states != null && !states.isEmpty()) {
+            String jpql = "SELECT aj FROM AsyncJobStatus aj WHERE aj.state IN (:states)";
+
+            return this.getEntityManager()
+                .createQuery(jpql, AsyncJobStatus.class)
+                .setParameter("states", states)
+                .getResultList();
+        }
+
+        return new LinkedList<>();
```

Comment:
It's not really that important, but why is this method returning a `LinkedList`, while the overloaded of this an `ArrayList` in the case no jobs were found? It is probably better for both to just return `Collections.emptyList()` since that is more consistent and cost effective (not creating a new object).

Reply:
I've made both return an ArrayList.

The reason I'm not using Collections.emptyList is due to the inconsistency we'd get with whether or not the list is mutable. If the list is returned from getResultList, then the list is mutable; otherwise it wouldn't be. I suppose we could wrap the query part in a Collections.unmodifiableList to make this method consistent, but then it will be the only curator that works this way. What're your thoughts here?

Reply:
No problem with not using emptyList since that would be inconsistent with other curators; the main issue was the ArrayList/LinkedList mismatch.